internal/services/api/reset_credentials.go

```diff
@@ -0,0 +1,52 @@
+package api
+
+// Copyright (C) 2023 by Posit Software, PBC.
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/posit-dev/publisher/internal/credentials"
+	"github.com/posit-dev/publisher/internal/logging"
+	"github.com/posit-dev/publisher/internal/types"
+)
+
+// We can ignore errors related to malformed data, we are resetting it ain't we?
+func errIsNotLoadError(err error) bool {
+	return err != nil && !errors.Is(err, &credentials.LoadError{}) && !errors.Is(err, &credentials.CorruptedError{})
+}
```

Comment:
We have this function in two spots - `provider_credentials` and here, can we combine them?